streaming/src/main/scala/io/gearpump/streaming/task/SerializerResolver.scala

```diff
@@ -0,0 +1,30 @@
+package io.gearpump.streaming.task
+
+import io.gearpump.esotericsoftware.kryo.util.ObjectMap
+import io.gearpump.esotericsoftware.kryo.util.IntMap
+import io.gearpump.streaming.task.SerializerResolver.Registration
+
+private[task] class SerializerResolver {
```

Comment:
Why it is `private`?

How do user to register for his own serializer? Can we provide user mechanism to register some new serializer for this solver? 

And further, is it possible to register some general serializer? E.g. `java.io.Serializable` or `hadoop.io.Writable` or kyro? That is, if there is a specific serializer, we can use it. otherwise, we can check whether there is any general serializer suitable for this message type? If so, we can use it as a last refugee instead of throw error.